### ISSUE TITLE
If a body is tagged with a MapIt ID then show a map of the area covered by the body

### ISSUE THREAD

**Issue by RichardTaylor** [mysociety/whatdotheyknow-theme#818]
eg. if 

https://www.whatdotheyknow.com/body/herefordshire_and_worcestershire_ccg

was tagged

mapit:163965

show a map of the area covered by the CCG on the CCG page. 

https://mapit.mysociety.org/area/163965.html

<img width="1239" alt="map-on-page" src="https://user-images.githubusercontent.com/547695/119505220-a5bdf100-bd64-11eb-9f18-f3c592ffda27.png">

Related: https://github.com/mysociety/alaveteli/issues/235 and https://github.com/mysociety/whatdotheyknow-theme/issues/579

**Comment by mdeuk** [comment]
**I love this idea** - the example you've given is one that'd also work extremely well for Councils, particularly as we look to improve our coverage of Town and Parish councils, where knowing the particular locus can be really, really helpful!

**Comment by mdeuk** [comment]
Just to add, I can see this also being very helpful for our regional pages - the enhanced metadata could allow for a more enriched experience to be generated there with less overall work, and it could also allow for better targetting of pro user requests (as an example: you might wish to make a FOI request to schools in a given local authority area, or to NHS trusts [bearing in mind there is often more than one]).

#416 #715 #716 #717 #747

**Comment by garethrees** [comment]
Just linking to an [email thread](https://groups.google.com/a/mysociety.org/g/alaveteli-dev-private/c/nBbJxj705wo/m/IDItaxXvAQAJ) about the Mapit aspect of this.

There's no embed option in Mapit, but would be relatively easy to render by the sounds of things:

```js
// The minimal HTML you'd need to replicate the map on WDTK would be
// something like: add the Leaflet CSS to the head of the page, and the
// Leaflet JS to the bottom, stick a <div id="map"></div> in the HTML
// with some styling to give it a width/height, then have a bit of extra
// JavaScript that does the same as MapIt to plot the area:

// Set up map with OSM background mapping
var map = new L.Map("leaflet");
map.attributionControl.setPrefix('');
var osm = new
  L.TileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
    attribution: 'Map © <a
    href="https://www.openstreetmap.org/copyright">OpenStreetMap</a>
    contributors',
    maxZoom: 18
  });
map.addLayer(osm);

// Fetch the GeoJSON of the area and add it to the map
var mapit = mapit || {};
mapit.area_loaded = function(data) {
  var area = new L.GeoJSON(data);
  // Leaflet double click on an area doesn't work by default
  area.on('dblclick', function(e){
    var z = map.getZoom() + (e.originalEvent.shiftKey ? -1 : 1);
    map.setZoomAround(e.containerPoint, z);
  });
  mapit.areas.addLayer(area);
  map.fitBounds(mapit.areas.getBounds());
};
mapit.areas = L.featureGroup().addTo(map);
reqwest({
  url: 'https://mapit.mysociety.org/area/163965.geojson?simplify_tolerance=0.0001',
  type: 'json',
  success: mapit.area_loaded
});
```

**Comment by RichardTaylor** [comment]
Maps would improve accessibility of the service helping us serve those who are not aware of, or easily able to research, relevant administrative boundaries. 

Maps might help ensure requests are directed to the correct body. 

Adding more geographical information might underpin geographical based features in the future eg. alerts for requests relating to a particular area or showing bodies which cover a particular area. https://github.com/mysociety/alaveteli/issues/235

**Comment by mdeuk** [comment]
> Maps would improve accessibility of the service helping us serve those who are not aware of, or easily able to research, relevant administrative boundaries.
> 

A rather similar scenario, which I've just described to volunteers@ would be the listing of schools within a particular geographic area. Richard and William have been doing excellent work adding schools to the database, so this would be complimentary.

Linking this with #908 would allow us to present a listing of bodies, say schools by type, within a relevant geography (say, at Country level, then County or City and so on).

This would also allow us to present the Scottish example, whereby the [majority of schools](https://education.gov.scot/education-scotland/scottish-education-system/policy-for-scottish-education/local-authority-role-and-duties/) [1] are under local authority control, and therefore we direct FoISA / EI(S)R requests to the local authority, as they are the relevant public body.

[1] The exception being those with "grant-aided" or "independent special school" status, which are public bodies in their own right, and are [explicitly covered under FoISA](https://www.itspublicknowledge.info/YourRights/WhocanIask/Organisations_designated_by_Scottish_Ministers.aspx); and the "independent" (private) schools which aren't included (but, maybe they should be?).

**Comment by RichardTaylor** [comment]
Map data appears to be available for Areas of Outstanding Natural Beauty 

https://github.com/mysociety/whatdotheyknow-theme/issues/988#issuecomment-1040502185

and National Parks

https://naturalengland-defra.opendata.arcgis.com/datasets/d333c7529754444894e2d7f5044d1bbf/explore

We list, or want to list, [the boards which run the Areas of Outstanding Natural Beauty ](https://www.whatdotheyknow.com/body/list/aonbcb)and [National Park Authorities](https://www.whatdotheyknow.com/body/list/npa). 

Maps showing the scope of these areas would help users.

**Comment by RichardTaylor** [comment]
Linking https://github.com/mysociety/mapit/issues/383

**Comment by RichardTaylor** [comment]
It would be great if we could have a map on a category page where all the bodies cover different areas eg. 

https://www.whatdotheyknow.com/body/list/nhs111

as well as maps on individual bodies. 

We could also use maps to [show where we're missing email addresses for bodies](https://github.com/mysociety/alaveteli/issues/3477), which could help gamify finding them.

**Comment by FOIMonkey** [comment]
If embedding maps is currently too hard, for now, we could link to the MapIt page from the sidebar on the body page, where we link to the data protection/charity registration.

This looks pretty trivial to do, unless the tags also need defining somewhere other than at https://github.com/mysociety/whatdotheyknow-theme/blob/master/lib/views/public_body/_more_info.html.erb

I have tagged all the IOW parish councils with the correct MapIt id number to allow us to experiment.

**Comment by FOIMonkey** [comment]
I've tagged all councils with the correct MapIt ID number in the form. mapit:id eg 

mapit:152125 -> https://mapit.mysociety.org/area/152125.html

We should be able to at least link to maps of council areas now.

**Comment by FOIMonkey** [comment]
I've not done the combined authorities, because I couldn't figure out how they were labelled. If anyone knows, there is a spreadsheet called rosetta stone on the drive that links up various bits of mySoc data, so please add them to there.

**Comment by FOIMonkey** [comment]
I've tagged 5949 parish and community councils with the correct MapIt id. I've also tagged these with the mapit id of the parent district council in the form of parent_mapit:id eg

parent_mapit:2595 -> https://mapit.mysociety.org/area/2595.html

This has the beneficial side effect of being able to view parish listings by district eg:
 
https://www.whatdotheyknow.com/body?tag=parent_mapit%3A2595

**Comment by garethrees** [comment]
Spiked a quick improvement to make these a bit more usable for users https://github.com/mysociety/whatdotheyknow-theme/pull/1330.

**Comment by HelenWDTK** [comment]
Closing as stale, but have added the "we want to do this but can't yet label", so it is not forgotten. It would be amazing, but would need to be part of a much bigger piece of funded work.
